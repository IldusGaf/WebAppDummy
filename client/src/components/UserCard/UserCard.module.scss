@import '../styleConstants.scss';

.userCard {
    display: flex;
    box-sizing: content-box;
    background-color: $elementColor;
    border: 0.1rem solid $elementBorderColor;
    flex-direction: column;
    justify-content: flex-start;
    border-radius: $contentBorderRadius;
    margin: 0.4rem;
    padding: 1rem;
    &__user {
        display: flex;
        flex-direction: column;
        align-items: center;
        justify-content: center;
        margin-bottom: 0.4rem;
    }
    &__userImage {
        margin: 0;
    img {
        border-radius: 50%;
        height: 40px;
        width: 40px;
    }
}
    &__userDesc {
        display: flex;
        justify-content: center;
        margin-left: 0.5rem;
    }
    &__userName {
        display: flex;
        text-align: center;
    }
    &__userDate {
        color: gray;
    }
    &__image {
        object-fit: cover;
        background-color: $backColor;
        border-radius: $contentBorderRadius;
        margin-bottom: 0.3rem;
        @include mediaSizeImgUsers;
    }
    &__desc {
        color: inherit;
    }
    &Dark{
        @include darkContent;
    }
    a {
        color: inherit;
    }
    @include mediaSizeUsers;
}