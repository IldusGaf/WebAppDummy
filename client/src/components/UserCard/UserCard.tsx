// import { UserOutlined } from '@ant-design/icons';
// import { Avatar } from 'antd';
import { UserOutlined } from '@ant-design/icons';
import { Avatar } from 'antd';
import React, { useContext } from 'react';
import { Link } from 'react-router-dom';
import { ThemeContext } from '../../contexts/ThemeContext';
import { OwnerType } from '../../types/dummyAPIResponses';
import classes from './UserCard.module.scss';

export const UserCard = ({
  id,
  title,
  firstName,
  lastName,
  picture,
}:OwnerType) => {
  const themeContext = useContext(ThemeContext);
  return (
    <div className={`${classes.userCard} ${themeContext.darkTheme ? classes.userCardDark : ''}`}>
      <Link to={`/user/${id}`}>
        <div className={classes.userCard__user}>
          {picture ? <img className={classes.userCard__image} alt={id} src={picture} />
            : <Avatar icon={<UserOutlined />} size={96} />}
          <div className={classes.userCard__userDesc}>
            <span className={classes.userCard__userName}>
              {title && `${title}. `}
              {firstName}
              {' '}
              {lastName}
            </span>
          </div>
        </div>
      </Link>
    </div>
  );
};
