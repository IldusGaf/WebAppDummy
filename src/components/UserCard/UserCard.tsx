// import { UserOutlined } from '@ant-design/icons';
// import { Avatar } from 'antd';
import React, { useContext } from 'react';
import { ThemeContext } from '../../contexts/ThemeContext';
import { OwnerType } from '../../types/dummyAPIResponses';
import classes from './UserCard.module.scss';

export const UserCard = ({
  id,
  title,
  firstName,
  lastName,
  picture,
}:OwnerType) => {
  const themeContext = useContext(ThemeContext);
  return (
    <div className={`${classes.userCard} ${themeContext.darkTheme ? classes.userCardDark : ''}`}>
      <div className={classes.userCard__user}>
        {/* <Avatar icon={<UserOutlined />} size="large" /> */}
        <img className={classes.userCard__image} alt={id} src={picture} />
        <div className={classes.userCard__userDesc}>
          <span className={classes.userCard__userName}>
            {title}
            {'. '}
            {firstName}
            {' '}
            {lastName}
          </span>
        </div>
      </div>
    </div>
  );
};
