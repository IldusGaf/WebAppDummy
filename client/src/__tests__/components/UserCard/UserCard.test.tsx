import React from 'react';
import { shallow } from 'enzyme';
import { UserCard } from '../../../components/UserCard/UserCard';

describe('UserCard component', () => {
  test('should render component', () => {
    const wrapper = shallow(<UserCard />);
    expect(wrapper.find('div.userCard')).toHaveLength(1);
  });

  test('snapshot test', () => {
    const wrapper = shallow(<UserCard />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should render data', () => {
    const id = '216544654697498';
    const title = 'mr';
    const firstName = 'James';
    const lastName = 'Bond';
    const picture = 'pictureUrl';
    const wrapper = shallow(<UserCard
      title={title}
      firstName={firstName}
      lastName={lastName}
      picture={picture}
      id={id}
    />);
    expect(wrapper.find('span.userCard__userName').text()).toBe(`${title}. ${firstName} ${lastName}`);
    expect(wrapper.find('img.userCard__image').prop('src')).toBe(picture);
  });
});
