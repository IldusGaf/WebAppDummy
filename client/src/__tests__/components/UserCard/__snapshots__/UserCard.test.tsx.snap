// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCard component snapshot test 1`] = `
<div
  className="userCard "
>
  <Link
    to="/user/undefined"
  >
    <div
      className="userCard__user"
    >
      <Avatar
        icon={<ForwardRef(UserOutlined) />}
        size={96}
      />
      <div
        className="userCard__userDesc"
      >
        <span
          className="userCard__userName"
        >
           
        </span>
      </div>
    </div>
  </Link>
</div>
`;
